In [1]:
pip install statsmodels

  Obtaining dependency information for statsmodels from https://files.pythonhosted.org/packages/9a/b2/13833f94076dcc2709137fee92157ef5a0099a11ddff4f4b7cb301b35357/statsmodels-0.14.1-cp311-cp311-win_amd64.whl.metadata
  Obtaining dependency information for scipy!=1.9.2,>=1.4 from https://files.pythonhosted.org/packages/9a/25/5b30cb3efc9566f0ebeaeca1976150316353c17031ad7868ef46de5ab8dc/scipy-1.12.0-cp311-cp311-win_amd64.whl.metadata
     ---------------------------------------- 0.0/60.4 kB ? eta -:--:--
     ---------------------------------------- 0.0/60.4 kB ? eta -:--:--
     ------------------- ------------------ 30.7/60.4 kB 660.6 kB/s eta 0:00:01
     -------------------------------- ----- 51.2/60.4 kB 525.1 kB/s eta 0:00:01
     -------------------------------------- 60.4/60.4 kB 460.6 kB/s eta 0:00:00
  Obtaining dependency information for patsy>=0.5.4 from https://files.pythonhosted.org/packages/43/f3/1d311a09c34f14f5973bb0bb0dc3a6e007e1eda90b5492d082689936ca51/patsy-0.5.6-py2.p


[notice] A new release of pip is available: 23.2.1 -> 24.0
[notice] To update, run: python.exe -m pip install --upgrade pip


In [2]:
pip install scipy

Note: you may need to restart the kernel to use updated packages.



[notice] A new release of pip is available: 23.2.1 -> 24.0
[notice] To update, run: python.exe -m pip install --upgrade pip


In [1]:
import numpy as np

# TASK 1

<b>Q1.</b>Suppose a child psychologist claims that the average time working mothers spend talking to their children is at least 11 minutes per day. You conduct a random sample of 1000 working mothers and find they spend an average of 11.5 minutes per day talking with their children. Assume prior research suggests the population standard deviation is 2.3 minutes.Conduct a test with a level of significance of alpha = 0.05.

<B>ANSWER:-</B>

<b>Ho</b> = μ ≤ 11 minutes  (NULL Hypothesis)

<b>Ha</b> = μ > 11 minutes (Alternative Hypothesis)

<b>x̄</b> = 11.5 minutes

<b>n</b> = 1000

<b>σ</b> = 2.3 minutes

<b>α</b> = 0.05

<u><b>Statistical Test</b></u>
Since we have a sample of 1000, which is greater than 30, and we have a sample mean, standard deviation, and want to compare the sample mean to a population value ,so we can use Z-test.

<table style="border:true">
<td><b>Formula Z=(x̄ - μ)/(σ/√n)</b></td>
</table>

Where: 
<ul>
<li>x̄ = sample mean (11.5 minutes)</li>
<li>μ = population mean (11 minutes)</li>
<li>σ = population standard deviation (2.3 minutes)</li>
<li>n = sample size (1000).</li>
</ul>

In [2]:
import math

In [3]:
Z = (11.5-11)/(2.3/math.sqrt(1000))
Z

6.874516652539955


### Find Critical Value/P-Value:
Since we're conducting a one-tailed test (since we're testing if the average time is more than 11 minutes), we will look up the critical value for a z-score corresponding to an alpha level of 0.05 from z-distribution table.
<p><i>corresponds to an alpha level of 0.05</i></p>
<p><i>Critical value=1.645</i></p>
Note :- Rejection criteria is z-score>critical value, fail to accept Ho




## Conclusion:
Since our calculated z-score (6.87) is much greater than the critical value (1.645), we <b>reject the null hypothesis</b>. We have sufficient evidence to conclude that the average time working mothers spend talking to their children is more than 11 minutes per day .

# TASK 2

<b>Q2.</b> A coffee shop claims that their average wait time for customers is less than 5 minutes. To test this claim, a sample of 40 customers is taken, and their wait times are recorded. The sample mean wait time is found to be 4.6 minutes with a standard deviation of 0.8 minutes. Perform a hypothesis test at a significance level of 0.05 and determine whether there is enough evidence to support the coffee shop's claim.

<B>ANSWER:-</B>


<b>
H</b>o = μ ≥ 5 minute (Null Hypothesis)sv  in-1) = 39

<b>Ha</b> = μ < 5 minutes (Alternative Hypothesis)


<b>x̄</b> = 4.6 minutes


<b>n</b> = 40


<b>σ</b> = 0.8 minutes

<b>μ</b> = 5 minutes


<b>α</b> = 0.05(5%)


<b>n-1</b> = 39

<u><b>Statistical Test</b></u>
Here the population std is unknown, so we can use t-test for this.
If the sample size is small (typically <30) and the population standard deviation is unknown, the t-test is appropriate. However, for larger sample sizes (usually greater than 30), the t-test can also be used even if the population standard deviation is unknown due to the Central Limit Theorem, which states that the distribution of sample means approaches a normal distribution as the sample size increases.

<table style="border:true">
<td><b>Formula t_test_statistic=(x̄ - μ) / (σ / √n)</b></td>
</table>


Where: 
<ul>
<li>x̄ = sample mean (4.6 minutes)</li>
<li>μ = population mean (5 minutes)</li>
<li>σ = Sample standard deviation (0.8 minutes)</li>
<li>n = sample size (40).</li>
    <li>α = Significance level (0.05(5%)).</li>
<li>n-1 = Degree of freedom(39).</li>
</ul>

In [7]:
t_test_statistic=(4.6 - 5) / (0.8 / np.sqrt(40))
print('t_statistic:',t_test_statistic)

t_statistic: -3.162277660168382


<p>The function t.ppf(alpha, df) calculates a critical value from the t-distribution based on the provided significance level (alpha) and degrees of freedom (df).</p>
<p>The term "percent point function" (ppf) means it gives you a value from the distribution such that if you randomly select a value from the distribution, there's a certain probability it will be less than or equal to that value.
</p>
    <p>In simpler terms, it helps to find a specific value in the t-distribution where a certain percentage of the distribution falls below that value.</p>


In [11]:
#calculating the critical value with df 39 and alpha 0.05
from scipy.stats import t
critical_value = t.ppf(0.05, 39)
print("Critical value:", critical_value)



Critical value: -1.6848751194974


## Conclusion:
Since our calculated t_test(-3.16) is less than the critical t-value (-1.685), we <b>fail to accept the null hypothesis</b>.
There is enough evidence to support the claim that the average wait time for customers is less than 5 minutes at a significance level of 0.05.
 